apps/src/studio/blocks.js

```diff
@@ -3202,16 +3202,11 @@ exports.install = function (blockly, blockInstallOptions) {
   };
 
   generator.studio_saySpriteChoices = function () {
+    const sprite = JSON.stringify(String(this.getFieldValue('SPRITE') || '0'));
+    const value = JSON.stringify(String(this.getFieldValue('VALUE') || ' '));
+
     // Generate JavaScript for saying (choices version).
-    return (
-      "Studio.saySprite('block_id_" +
-      this.id +
-      "', " +
-      (this.getFieldValue('SPRITE') || '0') +
-      ', ' +
-      JSON.stringify(String(this.getFieldValue('VALUE') || ' ')) +
-      ');\n'
-    );
+    return `Studio.saySprite('block_id_${this.id}', ${sprite}, ${value});\n`;
```

Comment:
Interestingly the `sprite` has the opposite behavior as value... looks like the `SPRITE` value was actually allowed to be an integer and not a string. I think the `spriteIndex` in `saySprite` needs to be an integer.